src/compiler/pre-transform/index.ts

```diff
@@ -0,0 +1,221 @@
+import pkg from '@storybook/addon-svelte-csf/package.json' with { type: 'json' };
+import type { ImportDeclaration, VariableDeclaration } from 'estree';
+import type { Comment, Fragment, Root, Script, SvelteNode } from 'svelte/compiler';
+
+import { transformComponentMetaToDefineMeta } from '#compiler/pre-transform/codemods/component-meta-to-define-meta';
+import { transformExportMetaToDefineMeta } from '#compiler/pre-transform/codemods/export-const-to-define-meta';
+import { transformImportDeclaration } from '#compiler/pre-transform/codemods/import-declaration';
+import { transformLegacyStory } from '#compiler/pre-transform/codemods/legacy-story';
+import { transformTemplateToSnippet } from '#compiler/pre-transform/codemods/template-to-snippet';
+import { MissingModuleTagError } from '#utils/error/parser/extract/svelte';
+
+interface Params {
+  ast: Root;
+  filename?: string;
+}
+
+export async function codemodLegacyNodes(params: Params): Promise<Root> {
+  const { walk } = await import('zimmerframe');
+
+  const { ast, filename } = params;
+
+  interface State {
+    componentIdentifierName: ComponentIdentifierName;
+    componentMetaHtmlComment?: Comment;
+    defineMetaFromMeta?: VariableDeclaration;
+  }
+  const state: State = {
+    componentIdentifierName: {},
+  };
+  let transformedAst = walk(ast as SvelteNode | Script, state, {
+    _(_node, context) {
+      const { next, state } = context;
+
+      next(state);
+    },
+
+    Root(node, context) {
+      const { fragment, module, ...rest } = node;
+      const { state, visit } = context;
+
+      if (!module) {
+        throw new MissingModuleTagError(filename);
```

Comment:
Oh, that complicates the work a bit, I think.

Two solutions:
1. Codemod add the `context="module"` to the `<script>` tag.
   Might be very dangerous operation, because there might be some logic created by the user unrelated to the Storybook usage. For example `onclick` snippets for button tests, etc.
2. Codemod will extract the most important pieces:
    1. import declarations for this addon - `"@storybook/addon-svelte-csf"`
    1. import declaration of the component _(optional)_
    1. `export const meta`
  ... and create a separate `<script context="module">` above the instance tag, with the extracted AST nodes from the instance tag.

Reply:
I think option 2 sounds best, if you think it's doable.

This is crucial though, as I think most usage of the v4 API is with the instance script today.